OData/src/System.Web.OData/OData/Builder/NavigationPropertyExtensions.cs

```diff
@@ -46,6 +46,11 @@ internal static class NavigationPropertyExtensions
             IEnumerable<StructuralTypeConfiguration> derivedTypes = builder.DerivedTypes(configuration);
             foreach (var config in derivedTypes)
             {
+                if (path.OfType<Type>().Any(p => p == config.ClrType) == true)
```

Comment:
I think of this as a readability and consistency issue. The sugar short hand of omitting the Boolean compare is harder to read and inconsistent with the way that the false case is handled.